cdap-master-spi/src/main/java/io/cdap/cdap/master/spi/environment/spark/SparkEnvConfig.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.master.spi.environment.spark;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents environment specific spark configurations.
+ */
+public class SparkEnvConfig {
+  private final String masterBasePath;
+  private final Map<String, String> envConfigs;
+
+  public SparkEnvConfig(String masterBasePath, Map<String, String> envConfigs) {
+    this.envConfigs = Collections.unmodifiableMap(new HashMap<>(envConfigs));
+    this.masterBasePath = masterBasePath;
+  }
+
+  /**
+   * Returns spark master base path. This should be used to set spark master url:
+   * https://spark.apache.org/docs/latest/submitting-applications.html#master-urls
+   */
+  public String getMasterBasePath() {
+    return masterBasePath;
+  }
+
+  /**
+   * Returns additional environment specific spark configurations.
```

Comment:
are these all things to go into the spark config (things passed with --conf flag)? Would be good to clarify, as environment kind of sounds like environment variables.

Reply:
yea that is true. let me rename it